aim2dat/io/cp2k/restart.py

```diff
@@ -0,0 +1,40 @@
+"""
+Functions to read the restart file of CP2K.
+"""
+
+# Internal library imports
+from aim2dat.io.cp2k.legacy_parser import RestartStructureParser
+from aim2dat.io.utils import read_multiple, custom_open
+
+
+@read_multiple(r".*-1\.restart$")
+def read_optimized_structure(folder_path):
+    """
+    Read optimized structures from 'restart'-files.
+
+    Parameters
+    ----------
+    folder_path : str
+        Path to the folder containing the CP2K ouput-files.
+
+    Returns
+    -------
+    dict or list
+        Dictionary containing the structural information. In case of a farming job a list of
+        dictionaries is returned. In case several calculations have been run in the same folder a
+        nested dictionary is returned.
+    """
+    structures = {}
+    for file_p, file_n in zip(folder_path["file"], folder_path["file_name"]):
+        proj = "-".join(file_n.split("-")[:-1])
```

Comment:
Instead of splitting and joining, one could also use `rsplit` with a maximum of one split starting from the right.

```suggestion
        proj = file_n.rsplit("-", 1)[0]
```

I'd say this is a question of personal preference, so up to you :)